### PR TITLE
Made postgres optional

### PR DESCRIPTION
**Pull request by MarcelCoding**
Added the ability to use a separate deployed postgres instance.

### CODE REVIEW

**Comment by WilliButz** [comment]
Good point, but I'd prefer to just use the existing `createDatabase` option and put all postgresql related config behind `mkIf cfg.createDatabase` to keep things simple. I don't see a need to have the enable switch. If someone desires a more complex setup than an enabled postgresql service + default database, then they can just leave it disabled and set up everything as they please, including a remote postgresql instance.

**Comment by MarcelCoding** [comment]
So the breaking change would be ok?

File: module.nix [hunk 1]
```diff
@@ -121,7 +121,7 @@ in
           enable = true;
           port = 6379;
         };
-        postgresql = {
+        postgresql = mkIf cfg.createDatabase {
           enable = true;
           package = pkgs.postgresql_14;
           ensureDatabases = mkIf cfg.createDatabase [ "authentik" ];
```
Comment (WilliButz) [inline module.nix:127]: These inner `mkIf cfg.createDatabase` can be removed now that all of the postgresql service config is behind this check above:
```suggestion
          ensureDatabases = [ "authentik" ];
```
(also the one below, but I'm unable to reference it in this review suggestion :))

**Review by WilliButz — CHANGES_REQUESTED** [review]
Just these minor leftovers and then it's good to merge :)

File: module.nix (hunk 1 above)
Comment (MarcelCoding) [reply module.nix]: done

**Review by WilliButz — APPROVED** [review]

**Pull request merged**